packages/api/tests/endpoints/test-rules.js

```diff
@@ -2962,3 +2962,103 @@ test('del() does not remove from PostgreSQL if removing from Elasticsearch fails
     )
   );
 });
+
+test.serial('Creating, deleting, and creating a rule with the same name succeeds', async (t) => {
+  const {
+    collectionPgModel,
+    newRule,
+    providerPgModel,
+    rulePgModel,
+    testKnex,
+  } = t.context;
+
+  const fakeCollection = fakeCollectionFactory();
+  const fakeProvider = fakeProviderFactory({
+    encrypted: true,
+    privateKey: 'key',
+    cmKeyId: 'key-id',
+    certificateUri: 'uri',
+    createdAt: new Date(2020, 11, 17),
+    updatedAt: new Date(2020, 12, 2),
+  });
+
+  newRule.provider = fakeProvider.id;
+  newRule.collection = {
+    name: fakeCollection.name,
+    version: fakeCollection.version,
+  };
+
+  newRule.rule = {
+    type: 'kinesis',
```

Comment:
Hey @kwartrosa, your changelog entry says you're testing a `onetime` rule. Did you mean to test `onetime` or `kinesis`? Looking at the slack thread of the original user issue, it seems like the issue is with `SNS` based rules?